core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -521,16 +509,17 @@ private[spark] class BlockManager(
               }
```

Comment:
In L496 of the new code, where we put the values we read from disk back into memory, do we need to "upgrade" the lock? Will someone else be able to read the same block in the mean time, since we did not acquire the write lock here?